source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1436,4 +1441,12 @@ void RtFileIo::writeDefinitions(const FilePath& documentPath, const RtTokenVec&
     writeDefinitions(ofs, names, options);
 }
 
+void RtFileIo::writePrim(std::ostream& stream, const RtPrim& prim, const RtWriteOptions* options)
```

Comment:
In order to be fully sure that the given prim is part of the current stage you could give the prim as a path instead